app/auth_test.go

```diff
@@ -86,26 +88,26 @@ func TestAuthManager(t *testing.T) {
 			Cmdr: cmdr,
 		},
 		KeyStore: store.NewKeystore(ms, "key", "", false),
-	})
+	}).(*MenderAuthManager)
```

Comment:
Question:
Why not return the struct from `NewAuthManager`, as opposed to an interface. It returns the same type anyways, and the interface is still implemented. If you need the interface, just receive it with an interface variable.

Reply:
I can try this, it might be simpler. Returning an interface was preexisting, so I didn't change it, but it might be better style in this case.

Reply:
Turns out this is tricky because of the `WithDBus` function. It returns `AuthManager`, which means it can no longer be put in the variable which was declared to hold `*MenderAuthManager` (default when using `:=`). And we also cannot change `WithDBus` to return `*MenderAuthManager`, because then the struct would be part of the interface. The only workaround is to declare the variable explicitly to hold `AuthManager`, using `var`, but I think this makes the API worse, and it's better that this slight casting ugliness is kept in the tests only.

It was worth a shot!

Reply:
So @oleorhagen convinced me that it is possible to do this, only the implementation's return type needs to be changed, the interface can still use an interface return type. A nice feature I didn't know about which solves the problems I encountered (thanks for sticking with it @oleorhagen!)

However, due to the number of people depending on this PR right now, I'm going to merge it as is, since it passed, and will followup with a new PR to fix it.